enhancements/network/network-policy-audit-logging.md

```diff
@@ -0,0 +1,315 @@
+```
+
+#### Story 2
+
+As an Openshift user I want to control the rate of ACL logging to ensure I can extract the useful information invloving accept/reject actions in high hit rate scenarios. 
+
+#### Story 3 
+
+As an Openshift user I want to monitor in realtime traffic that is hitting any network policies I have implemented. 
+
+--- TBD in implementation (Realtime Monitering would be nice here)
+
+### Implementation Details/Notes/Constraints 
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+The following codebase changes will need to done before this feature becomes usable  
+
+- OVN Kubernetes 
+
+  - Move from a cluster wide namespace default deny port group to namespace specific one 
+  - Add logging flag and custom OVN meter to all created ACLs, see below 
+  ```
+  fmt.Sprintf("log=%v", aclLogging != ""),
+  fmt.Sprintf("severity=%s", getACLLoggingSeverity(aclLogging)), "meter=acl-logging",
+  ```
+- Cluster Network Operator 
+  
+  - Digest the `aclLoggingRateLimit` Flag and start OVN-K with the configured value 
+
+
+- Logging Stack 
+  
+  - We need a way to collect and present ACL logging from the OVN controller logs 
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+- How do we control who is able to see the audit logs. 
+   - Only Admins should be able to see the audit logs since it could help expose holes in a cluster's network policy structure 
+
+- Scale questions involving the addition of OVN meters to all created ACL's 
+  - Adding meters to each ACL does add a slightly larger overhead in OVN
+  - Based on Early upstream testing with logging both enabled and disabled there were negligable performance changes  
+
+## Design Details
+
+### Open Questions 
+
+1. How should the ACL logs be digested from the ovn-controller logs and presented to the user? 
+  - Currenlty in OCP API audit logging is dumped directly onto the Nodes, should ACL logging do the same? 
+  - We could leverage OCP's logging stack to pull the logs and present to the user with a Kiabana Dashboard, However the logging 
+          stack is an optional feature in an OCP cluster 
+
+2. Based on the solution to question 1., Should a must-gather collect these logs, or should they be accessible only at runtime? 
```

Comment:
Change to:
2. Based on the solution to question 1:
   - Should a must-gather collect these logs, or should they be accessible only at runtime?